api/src/main/java/org/infinispan/api/reactive/KeyValueStore.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.api.reactive;
+
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+import org.reactivestreams.Publisher;
+
+/**
+ * A Reactive Key Value Store provides a highly concurrent and distributed data structure, non blocking and using
+ * reactive streams.
+ * <p>
+ * TODO: Add more javadoc and examples
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @see <a href="http://www.infinispan.org">Infinispan documentation</a>
+ * @since 10.0
+ */
+public interface KeyValueStore<K, V> {
```

Comment:
I'd leave `getAndSave` and `getAndDelete` out of the API. Over the years I've not found them to be that useful really. Plus, this kind of operations can be build by composing `get()` together with save/delete. True, that `getAnd...` operations might perform better, but I'd only add them in future versions if really needed.

Reply:
ok